symbulate/results.py

```diff
@@ -330,61 +365,216 @@ def plot(self, type=None, alpha=None, normalize=True, jitter=False,
+                else:
+                    caxes = fig.add_axes([0, 0.1, 0.05, 0.57])
+                cbar = plt.colorbar(mappable=histo[3], cax=caxes)
+                caxes.yaxis.set_ticks_position('left')
+                cbar.set_label('Density')
+                caxes.yaxis.set_label_position("left")
+                new_labels = []
+                for label in caxes.get_yticklabels():
+                    new_labels.append(int(label.get_text()) / len(x))
+                caxes.set_yticklabels(new_labels)
+            elif 'density' in type:
+                res = np.vstack([x, y])
+                density = gaussian_kde(res)
+                xmax, xmin = max(x), min(x)
+                ymax, ymin = max(y), min(y)
+                Xgrid, Ygrid = np.meshgrid(np.linspace(xmin, xmax, 100),
+                                           np.linspace(ymin, ymax, 100))
+                Z = density.evaluate(np.vstack([Xgrid.ravel(), Ygrid.ravel()]))
+                den = ax.imshow(Z.reshape(Xgrid.shape), origin='lower', cmap='Blues',
+                          aspect='auto', extent=[xmin, xmax, ymin, ymax]
+                )
+                if 'marginal' not in type:
+                    caxes = fig.add_axes([0, 0.1, 0.05, 0.8])
+                else:
+                    caxes = fig.add_axes([0, 0.1, 0.05, 0.57])
+                cbar = plt.colorbar(mappable=den, cax=caxes)
+                caxes.yaxis.set_ticks_position('left')
+                cbar.set_label('Density')
+                caxes.yaxis.set_label_position("left")
+            elif 'tile' in type:
+                x_uniq = np.unique(x)
+                y_uniq = np.unique(y)
+                x_pos = range(len(x_uniq))
+                y_pos = range(len(y_uniq))
+                x_map = dict(zip(x_uniq, x_pos))
+                y_map = dict(zip(y_uniq, y_pos))
+                x = np.vectorize(x_map.get)(x)
+                y = np.vectorize(y_map.get)(y)
+                nums = len(x)
+
+                counts = count_var(list(zip(y, x)))
+                intensity = np.zeros(shape=(len(y_uniq), len(x_uniq)))
+                    
+                for key, val in counts.items():
+                    intensity[key] = val / nums
+                hm = ax.matshow(intensity, cmap='Blues', origin='lower', aspect='auto')
+                ax.xaxis.set_ticks_position('bottom')
+                ax.set_xticks(x_pos)
+                ax.set_yticks(y_pos)
+                ax.set_xticklabels(x_uniq)
+                ax.set_yticklabels(y_uniq)
+
+                if 'marginal' not in type:
+                    caxes = fig.add_axes([0, 0.1, 0.05, 0.8])
+                else:
+                    caxes = fig.add_axes([0, 0.1, 0.05, 0.57])
+                cbar = plt.colorbar(mappable=hm, cax=caxes)
+                caxes.yaxis.set_ticks_position('left')
+                cbar.set_label('Relative Frequency')
+                caxes.yaxis.set_label_position("left")
+            elif 'mixed-tile' in type:
```

Comment:
The code in this block is repetitive: (1) you write essentially the same code for x and y, and (2) a lot of the logic is exactly the same as for "tile" plots.

As soon as you feel like you are writing the exact same code more than once, you should think about factoring it out. In this case, I think almost all of the logic for tile and mixed-tile plots should be factored into plot.py.